src/plugins/default/HeartbeatPlugin.ts

```diff
@@ -18,9 +18,12 @@ export class HeartbeatPlugin implements IEventPlugin {
     clearInterval(this._intervalId);
 
     let user: IUserInfo = context.event.data['@user'];
-    if (user && user.identity) {
-      this._intervalId = setInterval(() => context.client.submitSessionHeartbeat(user.identity), this._interval);
-    }
+
+    this._intervalId = setInterval(() => {
+      if (user && user.identity) {
+        context.client.submitSessionHeartbeat(user.identity);
+      }
+    }, this._interval);
```

Comment:
I don't think this would solve the issue as the plugin would still have to be triggered by an event submission, and it would be good if that first submission had some kind of session id or user id so we could create a session server side. 

in the .net client we have an internal session identifier that could make this simpler but the problem would still be there.
Please get a hold of me on slack and we'll talk through your scenario some more.